search/search.md

```diff
@@ -113,6 +113,58 @@ grep "\","
 
 Macros can help turn long, complex queries into easy-to-remember shortcuts. See [the full macro documentation](#!search/macros.md) for more information.
 
+## Compound Queries
+
+You can combine multiple queries together as a single "compound" query in order to leverage multiple data sources, fuse data into another query, and simplify complex queries. Gravwell's compound query syntax is a simple sequence of in-order queries, with additional notation to create temporary resources that can be referenced in queries later in the sequence. 
+
+A compound query consists of a main query (the last query in a sequence), and one or more subqueries. The main query is written just like a normal query, while subqueries are always wrapped in the notation `@<identifier>{<query>}`. Queries are separated by `;`. 
+
+Subqueries generate named resources in the form of `@<identifier>`. These resources can be used as regular resources with any module that supports resources such as `lookup`, `ipexist`, ... Named resources are scoped to the compound query they exist in, and are ephemeral - they are only accessible to other queries in the compound query, and are deleted as soon as the query is completed. 
+
+For example, say we have both DNS query and IP-level connection data under the tags "dns" and "conns", and we want to filter connection data down to only connections that didn't first have a corresponding DNS query. We can use compound queries to enrich our first query with DNS data and filter.
+
+Let's start with the subquery:
+
+```
+tag=dns table query answers
+```
+
+This produces a table:
+
+![](compound-ex1.png)
+
+In the subquery, we simply create a table of all queries and answers in our DNS data. Since this is a subquery, we need to give it a name so later queries can reference its output, and wrap the query in braces. We'll call this subquery "dns":
+
+```
+@dns{tag=dns table query answers}
+```
+
+In the main query, we use our connection data, and use the `lookup` module to read from our subquery "@dns":
+
+```
+tag=conns lookup -r @dns SrcIP answers query 
+| require -s -v query 
+| table SrcIP DstIP SrcIPBytes DstIPBytes 
+```
+
+This query uses the `lookup` module to add the "query" value from our subquery to any entry in our conns data that has a SrcIP that matches a DNS answer. From there we use an inverse `require` module to filter down to entries that didn't match, and then simply create a table of our data. 
```

Comment:
the lookup module has a "strict" `-s` flag so that you don't need the `require` module, might be worth being very concise and showing the "fastest" queries for examples.  Verbosity has merits too, so your call.

Reply:
Changed example and language to just use lookup with -s and -v